src/main/java/mate/academy/repository/book/BookSpecificationProviderManager.java

```diff
@@ -0,0 +1,22 @@
+package mate.academy.repository.book;
+
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import mate.academy.model.Book;
+import mate.academy.repository.SpecificationProvider;
+import mate.academy.repository.SpecificationProviderManager;
+import org.springframework.stereotype.Component;
+
+@RequiredArgsConstructor
+@Component
+public class BookSpecificationProviderManager implements SpecificationProviderManager<Book> {
+    private final List<SpecificationProvider<Book>> bookSpecificationProviders;
+
+    @Override
+    public SpecificationProvider<Book> getSpecificationProvider(String key) {
+        return bookSpecificationProviders.stream()
+                .filter(b -> b.getKey().equals(key)).findFirst()
+                .orElseThrow(() -> new RuntimeException("Can't find orrect specification for key "
+                        + key));
```

Comment:
don't use short one-letter naming, use EntityNotFound/NoSuchElement instead of generic runtime exception